rewrite-python/src/main/java/org/openrewrite/python/style/Autodetect.java

```diff
@@ -0,0 +1,107 @@
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.python.style;
+
+import org.openrewrite.SourceFile;
+import org.openrewrite.Tree;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.Statement;
+import org.openrewrite.python.PythonIsoVisitor;
+import org.openrewrite.python.tree.Py;
+import org.openrewrite.style.NamedStyles;
+import org.openrewrite.style.Style;
+
+import java.util.*;
+
+public class Autodetect extends NamedStyles implements PythonStyle {
+    public Autodetect(UUID id, Collection<Style> styles) {
+        super(id,
+                "org.openrewrite.python.Autodetect",
+                "Auto-detected",
+                "Automatically detect styles from a repository's existing code.",
+                Collections.emptySet(),
+                styles);
+    }
+
+    public static Detector detector() {
+        return new Detector();
+    }
+
+    public static class Detector {
+        private final IndentStatistics indentStatistics = new IndentStatistics();
+        private final FindIndentVisitor indentVisitor = new FindIndentVisitor();
+
+        public void sample(SourceFile python) {
+            if (python instanceof Py.CompilationUnit) {
+                indentVisitor.visitNonNull(python, indentStatistics);
+            }
+        }
+
+        public Autodetect build() {
+            return new Autodetect(
+                    Tree.randomId(),
+                    Collections.singletonList(indentStatistics.getTabsAndIndentsStyle())
+            );
+        }
+    }
+
+    private static class IndentStatistics {
+        private final Map<Integer, Integer> countsByIndentSize = new HashMap<>();
+
+        private<T> T keyWithHighestCount(Map<T, Integer> counts) {
+            int max = Collections.max(counts.values());
+            return counts.entrySet()
+                    .stream()
+                    .filter(entry -> entry.getValue() == max)
+                    .findFirst().get().getKey();
+        }
+
+        public TabsAndIndentsStyle getTabsAndIndentsStyle() {
+            int mostPopularIndent = keyWithHighestCount(countsByIndentSize);
+            return new TabsAndIndentsStyle(false, 2, mostPopularIndent, 4, false);
+        }
+    }
+
+    private static class FindIndentVisitor extends PythonIsoVisitor<IndentStatistics> {
+        private int currentBlockIndent = 0;
+
+        private int countSpaces(String s) {
+            int withoutSpaces = s.replaceAll(" ", "").length();
+            return s.length() - withoutSpaces;
+        }
```

Comment:
Even though this implementation currently only supports very simple detection, I would already now align it with the implementation we have for Java (see `FindIndentJavaVisitor` and `IndentStatistics` classes), as we will soon require that anyway to detect more complex cases (e.g. tabs or mixed spaces and tabs). Also note that we should try to avoid using methods like `String#replaceAll()` which behind the scenes will instantiate a regex pattern, as the allocation cost will put the GC under a lot of pressure.